### DISCUSSION TITLE
[OKD 4.10.0 on KVM UPI] bootstrap never complete

### DISCUSSION THREAD

**Discussion by yurix1975** [okd-project/okd#1328] [General]
good evening,
i spent a lot of time in last days to setup an okd cluster on UPI powered by qemu/proxmox, i've followed all the indications i found
but without success.
at the end of all my tries, the okd is unable to start ingress-router pod cause it doesn' t know about route.openshift.io resource kind.
i convinced my sefl the problem is on the bootstrap node, but which?
in the journal of the bootstrap i found a lot of failure during creation of object in the boostrap kube.
for example: 
Sep 01 11:33:46 okd4-bootstrap bootkube.sh[4399]: Created "0000_00_cluster-version-operator_00_namespace.yaml" namespaces.v1./openshift-cluster-version -n
Sep 01 11:33:46 okd4-bootstrap cluster-bootstrap[4481]: Failed to create "0000_00_cluster-version-operator_01_adminack_configmap.yaml" configmaps.v1./admin-acks -n openshift-config: namespaces "openshift-config" not found

my setup use fedora-coreos-35.20220327.3.0-qemu.x86_64.qcow2 for all the vm beloging to the cluster, a debian 11 vm as haproxy,dns,dhcp and router.

this is my install-config.yml:

apiVersion: v1
baseDomain: okd.local
metadata:
  name: lab

compute:
- hyperthreading: Enabled
  name: worker
  replicas: 0

controlPlane:
  hyperthreading: Enabled
  name: master
  replicas: 3

networking:
  clusterNetwork:
  - cidr: 10.128.0.0/14 
    hostPrefix: 23 
  networkType: OVNKubernetes
  serviceNetwork: 
  - 172.30.0.0/16

platform:
  none: {}

fips: false

pullSecret: '{"auths":{"fake":{"auth":"aWQ6cGFzcwo="}}}' 
sshKey: 'ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDXMXpke3A

in attach the bootstrap node journal
[journalctl.out.gz](https://github.com/okd-project/okd/files/9469858/journalctl.out.gz)

any help ?

kind regards

**Comment by titou10titou10** [comment]
I also have an OKD deployed on proxmox (v7.2-7)
I started with OKD v4.6 and I'm now at the latest version (v4.11.0-0.okd-2022-08-20-022919). I had, for various reasons, to reinstall OKD from scratch. I didn't had particular problems to install/update the cluster

At high level, what steps did you do to install your cluster and what config did you do before starting?

> my setup use fedora-coreos-35.20220327.3.0-qemu.x86_64.qcow2 for all the vm beloging to the cluster, a debian 11 vm as haproxy,dns,dhcp and router.

What do you mean by this? OKD will install by itself the OS on all the masters and workers node.

At very high level, what I provisionned:
- two VMs that act as load balancers in front of OKD. I use AlmaLinux v8.6+nginx as OS for the two for this, one for the apps and the other one for the API/control plane (it could be only one combined VM). The nginx acts as a reverse proxy, routing connections to the boostartp+masters for the API LB, and to the nodes (infra) where the OKD ingress controller is installed . I "label" 2 worker nodes as "infra" and force the ingress operator to be deployed on "infra" nodes
- one of the previous load balancer VM is configured to act a a web server to serve FCOS 36.x images fetched by the OKD installer + the iPXE config files. The nginx "as a web server" is only used the time of the installation
- a well configured DNS server (bind9 directly installed on the debian root proxmox server)
- I configured my ASUS router to act as a DHCP server to provision IP addresses for the nodes used by the boostratp/installer. Previsously, I didn't did that. Instead I directly configured the ip adress for each node in a distinct ipxe config file. That implies the necessity to have one ipxe config file per node instead of one per type (bootstrap, workers, masters)
- 7 "empty" VM: 1 bootstrap + 3 masters +3 workers  (24G + 5 CPUs + 40G storage...), start order=ne0/scsi0 (Important)

And that's basically it.

I upload the initial FCOS images +  the ipxe config files generated by the okd_installer to the nginx web server on the LB VM
I start the boostrap VM in ixpe, then "dhcp" then "chain http://<lb vm nginx>/bs.ixpe". The boostrap installs itself by fetching the ipxe bootsrap ipxe config file and the FCOS images from the LB and then "updated" it directly it necessary from the FCOS download site
Then I start the master the same way (with the "master ipxe file") , and then the workers
When the install is finishced, stop the boostrap VM and remove the routing in the ngix LB VM to the boostrap VM

I guess this is the standard procedure to install OKD on a "bare metal" VM management OS, including proxmox/qemu

**Reply by yurix1975** [reply]
ciao!
when i said to use fedora-coreos-35.20220327.3.0-qemu.x86_64.qcow2, i mean:
1) i made a template:
/usr/sbin/qm create 31352203 --name fedoracores35 --cores 4 --memory 6144 --net0 virtio,bridge=vmbr0
/usr/sbin/qm importdisk 31352203 fedora-coreos-35.20220327.3.0-qemu.x86_64.qcow2 local
/usr/sbin/qm set 31352203 --scsihw virtio-scsi-pci --scsi0 local:31352203/vm-31352203-disk-0.raw
/usr/sbin/qm set 31352203 --boot c --bootdisk scsi0
/usr/sbin/qm template 31352203

2) i cloned the template this way:
/usr/sbin/qm clone 31352203 170 --name okd-bootstrap
/usr/sbin/qm set 170 --net0 virtio,bridge=vmbr0,macaddr=BE:75:66:0C:D6:3B --memory 8192
/usr/sbin/qm resize 170 scsi0 +22G
echo "args: -fw_cfg name=opt/com.coreos/config,file=/opt/vmhome/okd/okd4/bootstrap.ign" >> /etc/pve/qemu-server/170.conf

/usr/sbin/qm clone 31352203 171 --name okd-master-1 
/usr/sbin/qm set 171 --net0 virtio,bridge=vmbr0,macaddr=D2:F1:FC:0E:81:2E --memory 8192
/usr/sbin/qm resize 170 scsi0 +54G
echo "args: -fw_cfg name=opt/com.coreos/config,file=/opt/vmhome/okd/okd4/master.ign" >> /etc/pve/qemu-server/171.conf

and  so on...

my environment consist of:
1) a virtual network 192.168.13.0/24
2) a vm connected to the virtual network 192.168.13.0/24 and my home net 192.168.1.0/24 (bridged):
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 6a:29:e4:96:60:d5 brd ff:ff:ff:ff:ff:ff
    altname enp0s18
    inet 192.168.1.168/24 brd 192.168.1.255 scope global dynamic eth0
       valid_lft 49933sec preferred_lft 49933sec
    inet6 fe80::6829:e4ff:fe96:60d5/64 scope link 
       valid_lft forever preferred_lft forever
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether e6:f2:60:2a:f2:77 brd ff:ff:ff:ff:ff:ff
    altname enp0s19
    inet 192.168.13.254/24 brd 192.168.13.255 scope global eth1
       valid_lft forever preferred_lft forever
    inet6 fe80::e4f2:60ff:fe2a:f277/64 scope link 
       valid_lft forever preferred_lft forever

this machine act ad dns server, dhcp, haproxy, routing (it's the default gw on the 192.168.13.0/24 and yes it masquerades the traffic from 192.168.13.0/24 )

this setup was documented [https://github.com/okd-project/okd/issues/27](url)
thanks for your reply
